lib/wasix/src/snapshot/log_file.rs

```diff
@@ -0,0 +1,261 @@
+                path,
+                ft,
+                accessed,
+                created,
+                modified,
+                len,
+                data,
+            } => Self::UpdateFileSystemEntryV1 {
+                path: path.into_owned(),
+                ft,
+                accessed,
+                created,
+                modified,
+                len,
+                data: data.into_owned(),
+            },
+            SnapshotLog::Snapshot => Self::SnapshotV1,
+        }
+    }
+}
+
+impl<'a> From<SnapshotLogEntry> for SnapshotLog<'a> {
+    fn from(value: SnapshotLogEntry) -> Self {
+        match value {
+            SnapshotLogEntry::TerminalDataV1 { data } => Self::TerminalData { data: data.into() },
+            SnapshotLogEntry::UpdateMemoryRegionV1 { start, end, data } => {
+                Self::UpdateMemoryRegion {
+                    region: start..end,
+                    data: data.into(),
+                }
+            }
+            SnapshotLogEntry::CloseThreadV1 { id } => Self::CloseThread { id },
+            SnapshotLogEntry::SetThreadV1 {
+                id,
+                call_stack,
+                memory_stack,
+            } => Self::SetThread {
+                id: id,
+                call_stack: call_stack.into(),
+                memory_stack: memory_stack.into(),
+            },
+            SnapshotLogEntry::CloseFileDescriptorV1 { fd } => Self::CloseFileDescriptor { fd },
+            SnapshotLogEntry::OpenFileDescriptorV1 { fd, state } => Self::OpenFileDescriptor {
+                fd,
+                state: state.clone(),
+            },
+            SnapshotLogEntry::RemoveFileSystemEntryV1 { path } => {
+                Self::RemoveFileSystemEntry { path: path.into() }
+            }
+            SnapshotLogEntry::UpdateFileSystemEntryV1 {
+                path,
+                ft,
+                accessed,
+                created,
+                modified,
+                len,
+                data,
+            } => Self::UpdateFileSystemEntry {
+                path: path.into(),
+                ft: ft.clone(),
+                accessed,
+                created,
+                modified,
+                len,
+                data: data.into(),
+            },
+            SnapshotLogEntry::SnapshotV1 => Self::Snapshot,
+        }
+    }
+}
+
+struct State {
+    file: tokio::fs::File,
+    at_end: bool,
+}
+
+/// The LogFile snap shooter will write its snapshots to a linear journal
+/// and read them when restoring. It uses the `bincode` serializer which
+/// means that forwards and backwards compatibility must be dealt with
+/// carefully.
+///
+/// When opening an existing journal file that was previously saved
+/// then new entries will be added to the end regardless of if
+/// its been read.
+///
+/// The logfile snapshooter uses a 64bit number as a entry encoding
+/// delimiter.
+pub struct LogFileSnapShooter {
```

Comment:
I'm not very fan of calling something "SnapShooter", I think just `SnapshotCapturer` or `Snapshot` would be better

Reply:
Renamed here:
https://github.com/wasmerio/wasmer/pull/4263/commits/8f11ac78b00d3f07b774e776d20e54221dc9e33f